cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/StandaloneAppFabricServer.java

```diff
@@ -70,11 +70,12 @@ public StandaloneAppFabricServer(CConfiguration cConf,
                                    SystemArtifactLoader systemArtifactLoader,
                                    PluginService pluginService,
                                    PrivilegesFetcherProxyService privilegesFetcherProxyService,
+                                   AppVersionUpgradeService appVersionUpgradeService,
                                    RouteStore routeStore) {
     super(cConf, sConf, discoveryService, schedulerService, notificationService, hostname, handlers,
           metricsCollectionService, programRuntimeService, applicationLifecycleService,
           programLifecycleService, streamCoordinatorClient, servicesNames, handlerHookNames, namespaceAdmin,
-          systemArtifactLoader, pluginService, privilegesFetcherProxyService, routeStore);
+          systemArtifactLoader, pluginService, privilegesFetcherProxyService, appVersionUpgradeService, routeStore);
```

Comment:
We don't technically need it. But this extends AppFabricServer. We need to pass in null and then don't start appVersionUpgradeService if is null etc. And we need to figure out whether it is sdk/distributed for the upgradehttphandler etc. Let me know if we need a different behavior for both. In SDK, the threads will quickly complete anyway.